Aspire Budgeting/Views/Dashboard/DashboardCardView.swift

```diff
@@ -1,42 +1,133 @@
+                     availableTotal: cardViewItem.availableTotal,
+                     categories: cardViewItem.categories,
+                     tintColor: baseColor)
+  }
 
   var body: some View {
-    VStack {
-      if !self.expanded {
-        CollapsedCardView(categoryName: categoryName, totals: totals, categoryRows: categoryRows)
-      } else {
-        ExpandedCardView(categoryName: categoryName, totals: totals, categoryRows: categoryRows)
-      }
-    }.background(Color.white.opacity(0.07))
-      .cornerRadius(10)
-      .shadow(radius: 5)
-      .padding()
-      .gesture(TapGesture().onEnded { _ in
-        withAnimation {
-          self.expanded.toggle()
-        }
+    VStack(alignment: .leading) {
+      topRow
+      secondRow
+      progressBar
+      fourthRow
+    }.sheet(isPresented: $showDetails) {
+      CategoryDetailsView(cardDetails: self.expandedDetails)
+    }
+  }
+}
+
+extension DashboardCardView {
+  private var topRow: some View {
+    HStack {
+      ZStack {
+        Circle()
+          .fill(Color.white)
+          .opacity(0.2)
+          .frame(width: 32, height: 32)
+
+        Image("moneyBag")
+      }.padding(.horizontal, 10)
+
+      Text(cardViewItem.title)
+        .font(.nunitoBold(size: 18))
+        .foregroundColor(.white)
+
+      Spacer()
+
+      Button(action: {
+        self.showDetails.toggle()
+      }, label: {
+        Text("Details >")
+          .font(.karlaRegular(size: 12))
+          .lineSpacing(3)
+          .foregroundColor(.white)
+          .overlay(
+            RoundedRectangle(cornerRadius: 14)
+              .stroke(Color.white, lineWidth: 1)
+              .frame(width: 79, height: 28)
+          )
+          .padding()
+          .padding(.trailing)
       })
+
+    }.padding(.top, 5)
+  }
+
+  private var secondRow: some View {
+    HStack {
+      Text("AVAILABLE")
+        .foregroundColor(.white)
+        .font(.nunitoRegular(size: 13))
+        .lineSpacing(3)
+        .padding(.horizontal, 10)
+
+      Spacer()
+
+      Text("BUDGETED")
+        .foregroundColor(.white)
+        .font(.nunitoRegular(size: 13))
+        .lineSpacing(3)
+        .padding(.trailing)
+    }
+  }
+
+  private var progressBar: some View {
+    AspireProgressBar(barType: .detailed,
+                      shadowColor: baseColor,
+                      tintColor: baseColor,
+                      progressFactor: self.cardViewItem.progressFactor)
+      .padding(.horizontal, 10)
+  }
+
+  private var fourthRow: some View {
+    HStack {
+      Text(self.cardViewItem.availableTotal.stringValue)
+        .foregroundColor(.white)
+        .font(.nunitoRegular(size: 13))
+        .lineSpacing(3)
+        .padding(.horizontal, 10)
+
+      Spacer()
+
+      Text(self.cardViewItem.budgetedTotal.stringValue)
+        .foregroundColor(.white)
+        .font(.nunitoRegular(size: 13))
+        .lineSpacing(3)
+        .padding(.trailing)
+    }
   }
 }
 
+//MARK:- Internal Types
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/5d9f74b3b35e3fb9746b1333ffd07f18323321a5/Aspire Budgeting/Views/Dashboard/DashboardCardView.swift#L120"></span>MARK comment should be in valid format. e.g. ‘// MARK: …’ or ‘// MARK: - …’<br />
<code>mark</code> <code>DashboardCardView.swift:120</code></td>
    </tr>
  </tbody>
</table>